### PR TITLE
Switch to build-module targets for e2e, uptest, render and yamllint and pull out project-specifc settings

### PR DESCRIPTION
**Pull request by kaessert**
### Description of your changes

Switch to build-module provided targets for:
- e2e
- uptest
- render
- yamllint

Also pulled out project-specific setting into `project.mk` which makes the Makefile fully generic.

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Manual verification by running described targets

### CODE REVIEW

**Comment by kaessert** [comment]
Test1:
```make e2e UPTEST_SKIP_DELETE=true
16:31:08 [ .. ] installing up v0.34.0
16:31:24 [ OK ] installing up v0.34.0
16:31:25 [ .. ] Building package configuration-app-v0.8.0-11.gc89db02.dirty.xpkg for linux_arm64
xpkg saved to /Users/tobiaskasser/up/configuration-app/_output/xpkg/linux_arm64/configuration-app-v0.8.0-11.gc89db02.dirty.xpkg
16:31:26 [ OK ] Built package configuration-app-v0.8.0-11.gc89db02.dirty.xpkg for linux_arm64
16:31:26 [ .. ] deleting controlplane
Deleting cluster "uptest-configuration-app" ...
Deleted nodes: ["uptest-configuration-app-control-plane"]
16:31:30 [ OK ] deleting controlplane
16:31:30 [ .. ] setting up controlplane
Creating cluster "uptest-configuration-app" ...
 ✓ Ensuring node image (kindest/node:v1.30.0) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-uptest-configuration-app"
You can now use your cluster with:

kubectl cluster-info --context kind-uptest-configuration-app

Have a question, bug, or feature request? Let us know! https://kind.sigs.k8s.io/#community 🙂
16:31:44 [ .. ] setting kubectl context to kind-uptest-configuration-app
Switched to context "kind-uptest-configuration-app".
"crossplane-build-module" has been added to your repositories
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "ealenn" chart repository
...Successfully got an update from the "eks" chart repository
...Successfully got an update from the "crossplane-build-module" chart repository
...Successfully got an update from the "prometheus-community" chart repository
...Successfully got an update from the "bitnami" chart repository
Update Complete. ⎈Happy Helming!⎈
16:31:50 [ .. ] setting up crossplane core with args --enable-usages
NAME: crossplane
LAST DEPLOYED: Tue Oct  8 16:31:50 2024
NAMESPACE: upbound-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
By proceeding, you are accepting to comply with terms and conditions in https://licenses.upbound.io/upbound-software-license.html

✨ Thank you for installing Universal Crossplane!
🚀 Install packages from https://marketplace.upbound.io to get started!
16:31:51 [ .. ] patching Crossplane with dev sidecar
deployment.apps/crossplane patched
deployment.apps/crossplane condition met
pod/crossplane-5fb96ff896-z77jn condition met
16:32:17 [ OK ] patching Crossplane with dev sidecar
16:32:17 [ .. ] copying local xpkg cache to Crossplane pod
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-2.g6b3cdda.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-2.g6b3cdda.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.g3997bdf.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.g4634cc4.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.g4bcd020.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.g591f344.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.g59fe60b.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.g94d235d.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-3.ga978ba1.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-4.g8b526af.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-5.g1b20c25.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.7.0-6.gddf04bd.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.8.0-1.gcff6335.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.8.0-11.gc89db02.dirty.gz
xpkg contents extracted to /Users/tobiaskasser/up/configuration-app/_output/xpkg/cache/configuration-app-v0.8.0-2.gdd99196.dirty.gz
16:32:19 [ OK ] copying local xpkg cache to Crossplane pod
16:32:19 [ .. ] deploying configuration package configuration-app v0.8.0-11.gc89db02.dirty
configuration.pkg.crossplane.io/configuration-app created
16:32:20 [ OK ] deploying configuration package configuration-app v0.8.0-11.gc89db02.dirty
16:32:20 [ .. ] running automated tests
SKIP_DEPLOY_ARGO= KUBECTL=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/kubectl-v1.24.3 CHAINSAW=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/chainsaw-0.2.0 CROSSPLANE_CLI=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/crossplane-cli-v1.17.1 CROSSPLANE_NAMESPACE=upbound-system YQ=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/yq-v4.40.5 /Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/uptest-v0.13.0 e2e examples/app-claim.yaml --data-source="" --setup-script=test/setup.sh  --skip-update --skip-import --skip-delete --default-timeout=2400s || (echo `date +%H:%M:%S` [FAIL] && false)
2024/10/08 16:32:20 Skipping update step because the root resource does not exist
2024/10/08 16:32:20 Skipping update step because the skip-delete option is set to true
2024/10/08 16:32:20 Skipping import step because the skip-import option is set to true
2024/10/08 16:32:20 Written test files: /var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e
2024/10/08 16:32:20 Running chainsaw tests at /var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e
2024/10/08 16:32:20 Version: 0.2.0
2024/10/08 16:32:20 Loading default configuration...
2024/10/08 16:32:20 - Using test file: 00-apply.yaml
2024/10/08 16:32:20 - TestDirs [/var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e/case]
2024/10/08 16:32:20 - SkipDelete true
2024/10/08 16:32:20 - FailFast false
2024/10/08 16:32:20 - ReportFormat ''
2024/10/08 16:32:20 - ReportName 'chainsaw-report'
2024/10/08 16:32:20 - Namespace ''
2024/10/08 16:32:20 - FullName false
2024/10/08 16:32:20 - IncludeTestRegex ''
2024/10/08 16:32:20 - ExcludeTestRegex ''
2024/10/08 16:32:20 - ApplyTimeout 5s
2024/10/08 16:32:20 - AssertTimeout 30s
2024/10/08 16:32:20 - CleanupTimeout 30s
2024/10/08 16:32:20 - DeleteTimeout 15s
2024/10/08 16:32:20 - ErrorTimeout 30s
2024/10/08 16:32:20 - ExecTimeout 5s
2024/10/08 16:32:20 - Parallel 1
2024/10/08 16:32:20 - NoCluster false
2024/10/08 16:32:20 Loading tests...
2024/10/08 16:32:20 - apply (/var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e/case)
2024/10/08 16:32:20 Loading values...
2024/10/08 16:32:20 Running tests...
2024/10/08 16:32:20 === RUN   chainsaw
2024/10/08 16:32:20 === PAUSE chainsaw
2024/10/08 16:32:20 === CONT  chainsaw
2024/10/08 16:32:20 === RUN   chainsaw/apply
2024/10/08 16:32:20 === PAUSE chainsaw/apply
2024/10/08 16:32:20 === CONT  chainsaw/apply
2024/10/08 16:32:20     | 16:32:20 | apply | @setup                   | CREATE    | OK    | v1/Namespace @ chainsaw-selected-wren
2024/10/08 16:32:20     | 16:32:20 | apply | Run Setup Script         | TRY       | RUN   |
2024/10/08 16:32:20     | 16:32:20 | apply | Run Setup Script         | CMD       | RUN   |
2024/10/08 16:32:20         === COMMAND
2024/10/08 16:32:20         /Users/tobiaskasser/up/configuration-app/test/setup.sh
2024/10/08 16:32:39     | 16:32:39 | apply | Run Setup Script         | CMD       | LOG   |
2024/10/08 16:32:39         === STDOUT
2024/10/08 16:32:39         Running setup.sh
2024/10/08 16:32:39         Waiting until all configurations are healthy/installed...
2024/10/08 16:32:39         configuration.pkg.crossplane.io/configuration-app condition met
2024/10/08 16:32:39         configuration.pkg.crossplane.io/configuration-app condition met
2024/10/08 16:32:39         Waiting until all installed provider packages are healthy...
2024/10/08 16:32:39         provider.pkg.crossplane.io/crossplane-contrib-provider-helm condition met
2024/10/08 16:32:39         Waiting for all pods to come online...
2024/10/08 16:32:39         deployment.apps/crossplane condition met
2024/10/08 16:32:39         deployment.apps/crossplane-contrib-function-patch-and-transform-6a1ab24d2512 condition met
2024/10/08 16:32:39         deployment.apps/crossplane-contrib-provider-helm-239d2e74884b condition met
2024/10/08 16:32:39         deployment.apps/crossplane-rbac-manager condition met
2024/10/08 16:32:39         Waiting for all XRDs to be established...
2024/10/08 16:32:39         compositeresourcedefinition.apiextensions.crossplane.io/xapps.platform.upbound.io condition met
2024/10/08 16:32:39         Setting up helm provider config pointing to the local cluster...
2024/10/08 16:32:39         clusterrolebinding.rbac.authorization.k8s.io/provider-helm-admin-binding created
2024/10/08 16:32:39         providerconfig.helm.crossplane.io/uptest created
2024/10/08 16:32:39         Setting up fake mariadb connection secret...
2024/10/08 16:32:39         secret/configuration-app-mariadb created
2024/10/08 16:32:39     | 16:32:39 | apply | Run Setup Script         | CMD       | DONE  |
2024/10/08 16:32:39     | 16:32:39 | apply | Run Setup Script         | TRY       | DONE  |
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | TRY       | RUN   |
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | APPLY     | RUN   | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | CREATE    | OK    | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | APPLY     | DONE  | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | SCRIPT    | RUN   |
2024/10/08 16:32:39         === COMMAND
2024/10/08 16:32:39         /bin/sh -c echo "Runnning annotation script"
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | SCRIPT    | LOG   |
2024/10/08 16:32:39         === STDOUT
2024/10/08 16:32:39         Runnning annotation script
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | SCRIPT    | DONE  |
2024/10/08 16:32:39     | 16:32:39 | apply | Apply Resources          | TRY       | DONE  |
2024/10/08 16:32:39     | 16:32:39 | apply | Assert Status Conditions | TRY       | RUN   |
2024/10/08 16:32:39     | 16:32:39 | apply | Assert Status Conditions | ASSERT    | RUN   | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:32:49     | 16:32:49 | apply | Assert Status Conditions | ASSERT    | DONE  | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:32:49     | 16:32:49 | apply | Assert Status Conditions | TRY       | DONE  |
2024/10/08 16:32:49 --- PASS: chainsaw (0.00s)
2024/10/08 16:32:49     --- PASS: chainsaw/apply (29.18s)
2024/10/08 16:32:49 PASS
2024/10/08 16:32:49 Tests Summary...
2024/10/08 16:32:49 - Passed  tests 1
2024/10/08 16:32:49 - Failed  tests 0
2024/10/08 16:32:49 - Skipped tests 0
2024/10/08 16:32:49 Done.
2024/10/08 16:32:49 Skipping test 01-update.yaml
2024/10/08 16:32:49 Skipping test 02-import.yaml
2024/10/08 16:32:49 Skipping test 03-delete.yaml
16:32:49 [ OK ] running automated tests
```

**Comment by kaessert** [comment]
Test2:
``` make uptest
16:33:40 [ .. ] running automated tests
SKIP_DEPLOY_ARGO= KUBECTL=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/kubectl-v1.24.3 CHAINSAW=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/chainsaw-0.2.0 CROSSPLANE_CLI=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/crossplane-cli-v1.17.1 CROSSPLANE_NAMESPACE=upbound-system YQ=/Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/yq-v4.40.5 /Users/tobiaskasser/up/configuration-app/.cache/tools/darwin_arm64/uptest-v0.13.0 e2e examples/app-claim.yaml --data-source="" --setup-script=test/setup.sh  --skip-update --skip-import --skip-delete --default-timeout=2400s || (echo `date +%H:%M:%S` [FAIL] && false)
2024/10/08 16:33:40 Skipping update step because the root resource does not exist
2024/10/08 16:33:40 Skipping update step because the skip-delete option is set to true
2024/10/08 16:33:40 Skipping import step because the skip-import option is set to true
2024/10/08 16:33:40 Written test files: /var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e
2024/10/08 16:33:40 Running chainsaw tests at /var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e
2024/10/08 16:33:40 Version: 0.2.0
2024/10/08 16:33:40 Loading default configuration...
2024/10/08 16:33:40 - Using test file: 00-apply.yaml
2024/10/08 16:33:40 - TestDirs [/var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e/case]
2024/10/08 16:33:40 - SkipDelete true
2024/10/08 16:33:40 - FailFast false
2024/10/08 16:33:40 - ReportFormat ''
2024/10/08 16:33:40 - ReportName 'chainsaw-report'
2024/10/08 16:33:40 - Namespace ''
2024/10/08 16:33:40 - FullName false
2024/10/08 16:33:40 - IncludeTestRegex ''
2024/10/08 16:33:40 - ExcludeTestRegex ''
2024/10/08 16:33:40 - ApplyTimeout 5s
2024/10/08 16:33:40 - AssertTimeout 30s
2024/10/08 16:33:40 - CleanupTimeout 30s
2024/10/08 16:33:40 - DeleteTimeout 15s
2024/10/08 16:33:40 - ErrorTimeout 30s
2024/10/08 16:33:40 - ExecTimeout 5s
2024/10/08 16:33:40 - Parallel 1
2024/10/08 16:33:40 - NoCluster false
2024/10/08 16:33:40 Loading tests...
2024/10/08 16:33:40 - apply (/var/folders/h4/sh6rlg_53s17_cjkfp90v3r00000gn/T/uptest-e2e/case)
2024/10/08 16:33:40 Loading values...
2024/10/08 16:33:40 Running tests...
2024/10/08 16:33:40 === RUN   chainsaw
2024/10/08 16:33:40 === PAUSE chainsaw
2024/10/08 16:33:40 === CONT  chainsaw
2024/10/08 16:33:40 === RUN   chainsaw/apply
2024/10/08 16:33:40 === PAUSE chainsaw/apply
2024/10/08 16:33:40 === CONT  chainsaw/apply
2024/10/08 16:33:40     | 16:33:40 | apply | @setup                   | CREATE    | OK    | v1/Namespace @ chainsaw-polite-glider
2024/10/08 16:33:40     | 16:33:40 | apply | Run Setup Script         | TRY       | RUN   |
2024/10/08 16:33:40     | 16:33:40 | apply | Run Setup Script         | CMD       | RUN   |
2024/10/08 16:33:40         === COMMAND
2024/10/08 16:33:40         /Users/tobiaskasser/up/configuration-app/test/setup.sh
2024/10/08 16:33:41     | 16:33:41 | apply | Run Setup Script         | CMD       | LOG   |
2024/10/08 16:33:41         === STDOUT
2024/10/08 16:33:41         Running setup.sh
2024/10/08 16:33:41         Waiting until all configurations are healthy/installed...
2024/10/08 16:33:41         configuration.pkg.crossplane.io/configuration-app condition met
2024/10/08 16:33:41         configuration.pkg.crossplane.io/configuration-app condition met
2024/10/08 16:33:41         Waiting until all installed provider packages are healthy...
2024/10/08 16:33:41         provider.pkg.crossplane.io/crossplane-contrib-provider-helm condition met
2024/10/08 16:33:41         Waiting for all pods to come online...
2024/10/08 16:33:41         deployment.apps/crossplane condition met
2024/10/08 16:33:41         deployment.apps/crossplane-contrib-function-patch-and-transform-6a1ab24d2512 condition met
2024/10/08 16:33:41         deployment.apps/crossplane-contrib-provider-helm-239d2e74884b condition met
2024/10/08 16:33:41         deployment.apps/crossplane-rbac-manager condition met
2024/10/08 16:33:41         Waiting for all XRDs to be established...
2024/10/08 16:33:41         compositeresourcedefinition.apiextensions.crossplane.io/xapps.platform.upbound.io condition met
2024/10/08 16:33:41         Setting up helm provider config pointing to the local cluster...
2024/10/08 16:33:41         clusterrolebinding.rbac.authorization.k8s.io/provider-helm-admin-binding created
2024/10/08 16:33:41         providerconfig.helm.crossplane.io/uptest unchanged
2024/10/08 16:33:41         Setting up fake mariadb connection secret...
2024/10/08 16:33:41         secret/configuration-app-mariadb unchanged
2024/10/08 16:33:41     | 16:33:41 | apply | Run Setup Script         | CMD       | DONE  |
2024/10/08 16:33:41     | 16:33:41 | apply | Run Setup Script         | TRY       | DONE  |
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | TRY       | RUN   |
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | APPLY     | RUN   | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | PATCH     | OK    | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | APPLY     | DONE  | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | SCRIPT    | RUN   |
2024/10/08 16:33:41         === COMMAND
2024/10/08 16:33:41         /bin/sh -c echo "Runnning annotation script"
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | SCRIPT    | LOG   |
2024/10/08 16:33:41         === STDOUT
2024/10/08 16:33:41         Runnning annotation script
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | SCRIPT    | DONE  |
2024/10/08 16:33:41     | 16:33:41 | apply | Apply Resources          | TRY       | DONE  |
2024/10/08 16:33:41     | 16:33:41 | apply | Assert Status Conditions | TRY       | RUN   |
2024/10/08 16:33:41     | 16:33:41 | apply | Assert Status Conditions | ASSERT    | RUN   | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:33:41     | 16:33:41 | apply | Assert Status Conditions | ASSERT    | DONE  | platform.upbound.io/v1alpha1/App @ default/configuration-app
2024/10/08 16:33:41     | 16:33:41 | apply | Assert Status Conditions | TRY       | DONE  |
2024/10/08 16:33:41 --- PASS: chainsaw (0.00s)
2024/10/08 16:33:41     --- PASS: chainsaw/apply (0.63s)
2024/10/08 16:33:41 PASS
2024/10/08 16:33:41 Tests Summary...
2024/10/08 16:33:41 - Passed  tests 1
2024/10/08 16:33:41 - Failed  tests 0
2024/10/08 16:33:41 - Skipped tests 0
2024/10/08 16:33:41 Done.
2024/10/08 16:33:41 Skipping test 01-update.yaml
2024/10/08 16:33:41 Skipping test 02-import.yaml
2024/10/08 16:33:41 Skipping test 03-delete.yaml
16:33:41 [ OK ] running automated tests
```

**Comment by kaessert** [comment]
Test4:
```make yamllint
16:39:44 [ .. ] running yamllint
16:39:45 [ OK ] running yamllint
```

**Review by ytsarev — COMMENTED** [review]
Looks great but I am missing some form of embedded documentation on how to run uptest.

Also instead of `Test{1,2,3,4}` it would be great to see the commands that were invoked for the testing , not only the output of the commands :)

File: project.mk [hunk 1]
```diff
@@ -0,0 +1,5 @@
+PROJECT_NAME := configuration-app
+UPTEST_INPUT_MANIFESTS := examples/app-claim.yaml
+UPTEST_SKIP_IMPORT := true
+UPTEST_SKIP_UPDATE := true
+UPTEST_SKIP_DELETE := true
```
Comment (ytsarev) [inline project.mk:5]: Is it overridable? Naively I would expect `UPTEST_SKIP_DELETE ?= true` here

Comment (kaessert) [reply project.mk:5]: Skip delete is supposed to be false here, let me correct that. And it is overrideable :)

**Comment by kaessert** [comment]
> Looks great but I am missing some form of embedded documentation on how to run uptest.
> 
> Also instead of `Test{1,2,3,4}` it would be great to see the commands that were invoked for the testing , not only the output of the commands :)

Copy paste issue, let me correct :) 
Docs will follow as well

File: apis/pat/composition.yaml
Comment (ytsarev) [inline apis/pat/composition.yaml:1]: why do we need this move to `apis/pat` ? it looks kind of confusing without alternative implementations

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,42 @@
...
+#		Claims need following annotations in order for render to work (adjust the paths
+#		if necessary):
+#			render.crossplane.io/composition-path: apis/pat/composition.yaml
+#       	render.crossplane.io/function-path: examples/functions.yaml
+#
+# - `e2e`
+#   Runs full end-to-end test, including creating cluster, setting up the configuration
+#   and testing if create, import and delete work as expected.
+#	Available options:
+#		SKIP_DELETE skips the deletion of any resources created during the test
```
Comment (ytsarev) [inline Makefile:24]: we should probably mention that SKIP_DELETE is false by default. Similar to the `SKIP_IMPORT` and `SKIP_UPDATE` below
https://github.com/crossplane/build/blob/main/makelib/uptest.mk#L23-L26

**Review by ytsarev — COMMENTED** [review]
overall looks great, left couple of comments

File: Makefile (hunk 2 above)
Comment (kaessert) [reply Makefile:24]: good point

File: apis/pat/composition.yaml
Comment (kaessert) [reply apis/pat/composition.yaml:1]: Let's keep it unified everywhere to be prepared for multi-language

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,42 @@
...
+#
+# - `e2e`
+#   Runs full end-to-end test, including creating cluster, setting up the configuration
+#   and testing if create, import and delete work as expected.
+#	Available options:
+#		SKIP_DELETE (default `false`) skips the deletion of any resources created during the test
+#		SKIP_UPDATE (default `false`) skips testing the update of the claims
+#		SKIP_IMPORT (default `true`) skips testing the import of resources
+#	Example:
+#		`make e2e SKIP_DELETE=true`
```
Comment (ytsarev) [inline Makefile:28]: Current instruction is wrong. We need to fix it together with Available options above
```suggestion
#		`make e2e UPTEST_SKIP_DELETE=true`
```

File: Makefile [hunk 4]
```diff
@@ -1,5 +1,42 @@
...
+#   Runs yamllint for all files in `api`-folder recursively
+#
+# - `render`
+#   Runs crossplane render to render the output of the composition. Usefule for quick
+#   feedback in order to test templating.
+#   Important note:
+#		Claims need following annotations in order for render to work (adjust the paths
+#		if necessary):
+#			render.crossplane.io/composition-path: apis/pat/composition.yaml
+#       	render.crossplane.io/function-path: examples/functions.yaml
```
Comment (ytsarev) [inline Makefile:18]: Consistent indentation
```suggestion
#			render.crossplane.io/composition-path: apis/pat/composition.yaml
#       	        render.crossplane.io/function-path: examples/functions.yaml
```

**Review by ytsarev — CHANGES_REQUESTED** [review]
Just doc issues to fix, otherwise stuff works 👍

File: Makefile (hunk 3 above)
Comment (kaessert) [reply Makefile]: fixed

File: Makefile (hunk 4 above)
Comment (kaessert) [reply Makefile]: fixed

**Review by ytsarev — APPROVED** [review]
Great reusable setup, thank you!

**Comment by kaessert** [comment]
@ytsarev Thank you for the thorough review!

**Pull request merged**